functions/src/services/GoogleSheets.ts

```diff
@@ -197,6 +232,59 @@ export default class GoogleSheets {
     return { spreadsheetId, updatedRows };
   }
 
+  /**
+   * Adds a list of data to specific row indices.
+   * First it adds empty rows, then it fills these empty rows
+   * with the provided list of data.
+   * @param rows Array of objects, each holds the data
+   *  to be written into a specific row
+   * @param ranges Array of row indices, where the data should be placed
+   */
+  public async addRows(rows: String[], ranges: number[]): Promise<any> {
```

Comment:
I'm not sure this input data structure is good. What if the arrays will have different length?

Maybe we can use [Sparse Arrays](http://2ality.com/2012/06/dense-arrays.html) for that?
https://stackoverflow.com/a/1510842